### PR TITLE
Add layer property

### PR DESCRIPTION
**Pull request by lhs8928**
### Change Description ###

Describe your changes here.

For instance,
 - Fix broken links of native application docs.
 -  ...

### Bugs Fixed ###

Provide links to bugs here

For instance,

 - Issue #265

### API Changes ###

In case of ACR, put the link of ACR

For instance,

 - ACR: ACR-1120

### CODE REVIEW

File: docs/application/native/guides/machine-learning/machine-learning-training.md [hunk 1]
```diff
@@ -108,67 +108,193 @@ ml_train_set_property(layer, "unit=10", "activation=softmax", "bias_initializer=
 ml_train_model_add_layer(model, layer);
 ```
 
-Following are the available properties for each layer type:
+There are 2 types of layers which includes (weight_initializer, bias_initializer, weight_regularizer, weight_regularizer_constant) properties and a layer which doesn't includes.
```
Comment (zhoonit) [inline docs/application/native/guides/machine-learning/machine-learning-training.md]: does not

Comment (zhoonit) [reply docs/application/native/guides/machine-learning/machine-learning-training.md]: which includes commonly trainable weights,

File: docs/application/native/guides/machine-learning/machine-learning-training.md [hunk 2]
```diff
@@ -108,67 +108,193 @@ ml_train_set_property(layer, "unit=10", "activation=softmax", "bias_initializer=
...
+&#xfeff;                                                     |                             | tanh                      | Hyperbolic tangent
+&#xfeff;                                                     |                             | sigmoid                   | Sigmoid function
+&#xfeff;                                                     |                             | relu                      | Relu function
+&#xfeff;                                                     |                             | softmax                   | Softmax function
+&#xfeff;                                                     | loss                        | (float)                   | Loss
+&#xfeff;                                                     | weight_initializer          | (categorical)             | Weight initializer
+&#xfeff;                                                     |                             | zeros                     | Zero initialization
+&#xfeff;                                                     |                             | lecun_normal              | LeCun Normal Initialization
```
Comment (zhoonit) [inline docs/application/native/guides/machine-learning/machine-learning-training.md]: consistency

File: docs/application/native/guides/machine-learning/machine-learning-training.md [hunk 3]
```diff
@@ -108,67 +108,193 @@ ml_train_set_property(layer, "unit=10", "activation=softmax", "bias_initializer=
...
+`ML_TRAIN_LAYER_TYPE_POOLING2D` (since 6.5)                  |                             |                           | Pooling layer
+&#xfeff;                                                     | pooling                     | (categorical)             | Pooling type
+&#xfeff;                                                     |                             | max                       | Max pooling
+&#xfeff;                                                     |                             | average                   | Average pooling
+&#xfeff;                                                     |                             | global_max                | Global max pooling
+&#xfeff;                                                     |                             | global_average            | Global average pooling
+&#xfeff;                                                     | pool_size                   | (unsigned integer vector) | Comma-seperated unsigned intergers for pooling size, `height, width`  respectively
```
Comment (zhoonit) [inline docs/application/native/guides/machine-learning/machine-learning-training.md]: omit vector, array

File: docs/application/native/guides/machine-learning/machine-learning-training.md [hunk 4]
```diff
@@ -108,67 +108,193 @@ ml_train_set_property(layer, "unit=10", "activation=softmax", "bias_initializer=
...
+&#xfeff;                                                     |                             | l2norm                    | L2 weight regularizer
+&#xfeff;                                                     | weight_regularizer_constant | (float)                   | Weight regularizer constant
+`ML_TRAIN_LAYER_TYPE_FC`                                     |                             |                           | Fully connected layer
+&#xfeff;                                                     | unit                        | (unsigned integer)        | Number of outputs
+`ML_TRAIN_LAYER_TYPE_CONV2D` (since 6.5)                     |                             |                           | 2D Convolution layer
+&#xfeff;                                                     | filters                     | (unsigned integer)        | Number of filters
+&#xfeff;                                                     | kerner_size                 | (unsigned integer array)  | Comma-seperated unsigned integers for kernel size, `height, width`  respectively
```
Comment (zhoonit) [inline docs/application/native/guides/machine-learning/machine-learning-training.md]: Array of unsigned integer

Comment (zhoonit) [inline docs/application/native/guides/machine-learning/machine-learning-training.md]: kernel_size

File: docs/application/native/guides/machine-learning/machine-learning-training.md [hunk 5]
```diff
@@ -108,67 +108,193 @@ ml_train_set_property(layer, "unit=10", "activation=softmax", "bias_initializer=
...
+&#xfeff;                                                     | stride                      | (unsigned integer array)  | Comma-seperated unsigned integers for strides, `height, width`  respectively
+&#xfeff;                                                     | padding                     | (categorical)             | Padding type
+&#xfeff;                                                     |                             | valid                     | No padding
+&#xfeff;                                                     |                             | same                      | Preserve height/width dimension
+&#xfeff;                                                     |                             | (unsigned integer)        | Padding for all 4 direction
```
Comment (zhoonit) [inline docs/application/native/guides/machine-learning/machine-learning-training.md]: size of padding applied uniformly to all side

File: docs/application/native/guides/machine-learning/machine-learning-training.md [hunk 6]
```diff
@@ -108,67 +108,193 @@ ml_train_set_property(layer, "unit=10", "activation=softmax", "bias_initializer=
...
+&#xfeff;                                                     |                             | (unsigned integer array)  | Padding for height, width
```
Comment (zhoonit) [inline docs/application/native/guides/machine-learning/machine-learning-training.md]: array of unsigned integer of size 2

File: docs/application/native/guides/machine-learning/machine-learning-training.md [hunk 7]
```diff
@@ -108,67 +108,193 @@ ml_train_set_property(layer, "unit=10", "activation=softmax", "bias_initializer=
...
-| &#xfeff;                         |                             | global_average       | global average pooling                                                |
-| `ML_TRAIN_LAYER_TYPE_FLATTEN`    |                             |                      | Flatten layer                                                         |
-| `ML_TRAIN_LAYER_TYPE_ACTIVATION` |                             |                      | Standalone, simple activation layer                                   |
-| `ML_TRAIN_LAYER_TYPE_ADDITION`   |                             |                      | Addition layer                                                        |
-| `ML_TRAIN_LAYER_TYPE_CONCAT`     |                             |                      | Concatenation layer                                                   |
-| `ML_TRAIN_LAYER_TYPE_MULTIOUT`   |                             |                      | Multiout layer, repeats output to the number of layers                 |
+(Universal properties)                                       |                             |                           | Universal properties that applies to every layer
```
Comment (zhoonit) [inline docs/application/native/guides/machine-learning/machine-learning-training.md]: We will need default value column

File: docs/application/native/guides/machine-learning/machine-learning-training.md [hunk 8]
```diff
@@ -244,29 +370,30 @@ Following is an example of the `*.ini` file:
...
 epsilon = 1e-7  # epsilon for adam
 
-#### Model compile related properties
-batch_size = 9
-loss = cross      # Cost(loss) function : cross entropy(cross)
-
-####  Model run related properties
-Epochs = 20     # Epochs
-save_path = "model.bin"   # model path to save and read parameters
+[Train_set]
```
Comment (zhoonit) [inline docs/application/native/guides/machine-learning/machine-learning-training.md]: train_set

File: docs/application/native/guides/machine-learning/machine-learning-training.md [hunk 9]
```diff
@@ -244,29 +370,30 @@ Following is an example of the `*.ini` file:
...
 
-[DataSet] # Special section that describes dataset
-BufferSize=9
-TrainData="trainingSet.dat"
-ValidData="validationSet.dat"
-LabelData="label.dat"
+[valid_set]
+BufferSize = 9
+Type = file
+path = "valSet.dat"
```
Comment (zhoonit) [inline docs/application/native/guides/machine-learning/machine-learning-training.md:398]: please comment below like
```
# Add [test_set] section if applicable,.
```

File: docs/application/native/guides/machine-learning/machine-learning-training.md [hunk 10]
```diff
@@ -108,67 +108,193 @@ ml_train_set_property(layer, "unit=10", "activation=softmax", "bias_initializer=
 ml_train_model_add_layer(model, layer);
 ```
 
-Following are the available properties for each layer type:
+There are 2 types of layers which includes commonly trainable weigths and a layer which does not includes.
```
Comment (zhoonit) [inline docs/application/native/guides/machine-learning/machine-learning-training.md]: which include

Comment (zhoonit) [reply docs/application/native/guides/machine-learning/machine-learning-training.md]:
```suggestion
There are 2 types of layers.
One type includes commonly trainable weights and another type does not include.
```

File: docs/application/native/guides/machine-learning/machine-learning-training.md [hunk 11]
```diff
@@ -108,67 +108,193 @@ ml_train_set_property(layer, "unit=10", "activation=softmax", "bias_initializer=
...
+The following are the available properties for each layer type which include (weight_initializer, bias_initializer, weight_regularizer, weight_regularizer_constant) properties.
```
Comment (zhoonit) [inline docs/application/native/guides/machine-learning/machine-learning-training.md]:
```suggestion
The following are the available properties for each layer type which include commonly trainable weights.
```
